src/prefect_server/api/artifacts.py

```diff
@@ -0,0 +1,86 @@
+from prefect import models
+from prefect.utilities.plugins import register_api
+
+
+@register_api("artifacts.create_task_run_artifact")
+async def create_task_run_artifact(
+    task_run_id: str,
+    kind: str,
+    data: dict,
+    tenant_id: str = None,
+) -> str:
+    """
+    Create a task run artifact.
+
+    Args:
+        - task_run_id (str): the task run id
+        - kind (str): the artifact kind
+        - data (dict): the artifact data
+        - tenant_id (str, optional): the tenant id that this artifact belongs to. Defaults
+            to the tenant ID linked to the task run
+
+    Returns:
+        - str: the task run artifact ID
+    """
+    if not task_run_id:
+        raise ValueError("A `task_run_id` must be provided to create artifact")
+
+    task_run = await models.TaskRun.where(id=task_run_id).first({"tenant_id"})
+
+    if not task_run:
+        raise ValueError(f"Task run {task_run_id} not found")
+
+    if tenant_id and tenant_id != task_run.tenant_id:
+        raise ValueError("Provided tenant ID does not match task run")
```

Comment:
```suggestion
        raise ValueError("Invalid tenant ID.")
```
We don't want to give the impression that you can validate a given tenant ID by spamming this route so instead simply saying "Invalid" is better